pkg/authorization/api/types.go

```diff
@@ -118,6 +118,33 @@ func init() {
 		Difference(NormalizeResources(sets.NewString(GroupsToResources[KubeEscalatingViewableGroupName]...))).List()
 }
 
+// we care argue exact shape later.  Point is, it's easily convertible to and from resourcequota
+type ClusterResourceQuota struct {
+	unversioned.TypeMeta
+	kapi.ObjectMeta
+
+	// Spec defines the desired quota
+	Spec ClusterResourceQuotaSpec
+
+	// Status defines the actual enforced quota and its current usage
+	Status kapi.ResourceQuotaStatus
+}
+
+type ClusterResourceQuotaSpec struct {
+	Selector map[string]string
```

Comment:
we should not allow a nil selector for the sake of not shooting ourselves in our foot.